webclient/core/templates/core/export_modal.html

```diff
@@ -48,55 +45,60 @@ <h5 class="modal-title">{% trans "core.templates.core.export-modal.title" %}</h5
+      const controller = new AbortController();
+      const signal = controller.signal;
+      
+      const cancelButton = document.getElementById('export-progress-modal-button-cancel');
+      cancelButton.onclick = () => {
+          controller.abort();
       }
 
-      document.addEventListener('DOMContentLoaded', function (event) {
-          document.getElementById('export-progress-modal-button').addEventListener('click', resetTempValue);
-          try {
-            document.getElementById('export-progress-modal-button-cancel').addEventListener('click', resetTempValue);
-          }
-          catch {
-            console.log("no element export-progress-modal-button-cancel")
-          }
-          document.getElementById('export-progress-modal-button-stop').addEventListener('click', () => {
-              window.stop()
-              enableCloseButton()
-              resetTempValue()
-          })
-      })
+      const manageExportDownload = (target_link) => {
+          const downloadUrl = target_link;
+          
+          const downloadStartEvent = new Event('downloadStart');
+          document.dispatchEvent(downloadStartEvent);
 
-      const checkValue = () => {
-          const cancelButton = document.getElementById("export-progress-modal-button-cancel")
-          if (cancelButton){
-              cancelButton.style.display = "none";
-          }
-          const stopButton = document.getElementById("export-progress-modal-button-stop")
-          if (stopButton){
-              stopButton.style.display = null;
-          }
-          const button = document.getElementById("export-progress-modal-button")
-          if (button){
-              button.style.display = null;
-          }
           const rawEmail = "{{ request.user.email }}"
           const email = rawEmail.replace("@", "(at)")
           fetch(`{% url 'core:read_temp_value' %}?temp_name=export_${email}`)
               .then(response => response.json())
               .then(data => {
                   updateProgressBar(data.value);
-                  if (data.value !== 100) {
-                      setTimeout(checkValue, 1000);
-                  } else {
-                      enableCloseButton();
-                      resetTempValue();
-                  }
               })
               .catch(error => console.error('Error in Check Value Function:', error));
-      }
+
+          fetch(downloadUrl, { signal })
+              .then(response => {
+                  if (!response.ok) throw new Error('Network response was not ok.');
+                  return response.blob();
+              })
+              .then(blob => {
+                  const downloadCompleteEvent = new Event('downloadComplete');
+                  document.dispatchEvent(downloadCompleteEvent);
+
+                  const downloadLink = document.createElement('a');
+                  downloadLink.href = URL.createObjectURL(blob);
+                  downloadLink.download = "export.csv";
```

Comment:
@pesikj Tady mě trochu mate ten název `export.csv` - může přece jít i o JSON nebo XLSX. Nebo je to jen nějaký placeholder?

Reply:
@motyc Kvůli potenciálnímu konfliktu vyřešeno [zde](https://github.com/ARUP-CAS/aiscr-webamcr/blob/5b2952231fd0ab7d17559f9d2a9c619fa30bf31d/webclient/core/templates/core/export_modal.html)